### ISSUE TITLE
vue/no-unused-vars reports error when the var is used in the child component.

### ISSUE THREAD

**Issue by adcentury** [vuejs/eslint-plugin-vue#690]
**Tell us about your environment**	
* **ESLint version:**	`^4.15.0`
* **eslint-plugin-vue version:**	`^4.0.0`
* **Node version:**	 `10.11.0`

**Please show your full configuration:**

```javascript
// https://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint'
  },
  env: {
    browser: true,
  },
  // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
  // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
  extends: ['plugin:vue/essential', 'airbnb-base'],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  globals: {
    "require": true
  },
  // check if imports actually resolve
  settings: {
    'import/resolver': {
      webpack: {
        config: 'build/webpack.base.conf.js'
      }
    }
  },
  // add your custom rules here
  rules: {
    // don't require .vue extension when importing
    'import/extensions': ['error', 'always', {
      js: 'never',
      vue: 'never'
    }],
    // disallow reassignment of function parameters
    // disallow parameter object manipulation except for specific exclusions
    'no-param-reassign': ['error', {
      props: true,
      ignorePropertyModificationsFor: [
        'state', // for vuex state
        'acc', // for reduce accumulators
        'e' // for e.returnvalue
      ]
    }],
    // allow optionalDependencies
    'import/no-extraneous-dependencies': ['error', {
      optionalDependencies: ['test/unit/index.js']
    }],
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    // others
    'no-new': 0,
    'max-len': 0,
    'no-else-return': 0,
    'arrow-body-style': 0,
    'func-names': 0,
    'default-case': 0,
    'no-alert': 0,
    'no-underscore-dangle': 0,
    'no-empty-pattern': 0,
    'no-restricted-syntax': 0,
    'guard-for-in': 0,
    'no-param-reassign': 0,
    'no-plusplus': 0,
    "vue/no-parsing-error": [2, {
      "x-invalid-end-tag": false,
    }],
  }
}

```

**What did you do?**	

```vue
<!-- this project uses iview components -->
<MenuItem
  v-for="(item, index) in someList"
  :key="item.name"
  ...
>
  <Row>
      <Col>
        <Tooltip :placement="getPlacementByIndex(index)">
        ...
</MenuItem>
```

**What did you expect to happen?**
Pass the lint.

**What actually happened?**

```
error  'index' is defined but never used  vue/no-unused-vars
```

**Comment by mysticatea** [comment]
Thank you for the report.

But I couldn't reproduce it on [our online demo](https://mysticatea.github.io/vue-eslint-demo/#eJyNVsuO2zgQ/BVBpwQwZ4A9Gk4Ou6cAu8BiN0dfKKklMaFIpUnZVgbz7ymKesuHHDwWq1rNrn6N39LcFpSe04unptXS0+erSZJLoW7DAx7/IdN9ARuPSXITpeVP1/SDAnhKlCno8RFfibMN/a2cv6aT6fk79bAMhi9GNjQxnyeDy3/2Ph8G4C+rNwCgr9Zqr9rkjPhyash4+KzI/zsd/+y/hCA+xFCu6c7By8vLKOV10hJFvkaVl9eV9ovLWbUeT/RoLfukoFJ22idvV/MebEc+PaXcaXLp+Q0pNM5zl3vLwnUtcXr+45QiS6JQTLlX1gwIYvbEgsl3HBFjRS4diYJyLVkGSzcTWjon8FHVYmybVjIJQ5X4SWwX3BQHU+OspvXRPzHx0vjhdTWHGRnPViPUih4TWFDWVdWobjhr8iRuckEgXkiu3AaIQhpqMuItg/ZYA1rlKMKQkQlFYXwv8hrJyUPuBl9bspUhq2YDzofHTjI9PEuRoaNImnCT3zKOGjUhpdTa12y7qp6hzuQ7j5W2mdQ7UBmDYJ9VVZmb1KqImW1nlHHuYCzuNabFtejsiWvUgwoxQGgH1NnLbHZn6C5c32RWT4jNvkGX1rOJzb2cWaYYFC0A4hi9T5gjXe4EwYBdqC3LpWS+Vk5khEanVd+D6DCI5XKAUEwBRDQYJKWVoYVjkqhupleQkyWJUhmI6Hco+l6u+7QznYNfLTNaBI8g+nKBIInQhORy2carkCcEJVyNIa+R1gFk+tFhZEWvSEcEbwrljIxXxtr5viUb9d06esVQhi00DfuNZuabe4Sr3RJLQBGPdD0aSZlhoLuQmpYtEugVbewwh2XnlKnEsHXxV5VrfjtFCzhOEiaDVQb/Gz6UMrgkZstrgskR3xDM3qPD/AEOwVozSJVebgxUQYLKEvrdb8ia9m24aIs/PG6SaBQsprymNTmWdQ7CPSH3eQ7VCzkTd+XrmMOZnmq9qFLuQIb4h3qL1ZI/WDGh4TeLfc3G0q2lzszQTuN/mOGulUnw9SyXi5uxG6dksrV+x91EpsbWXoO5tvL7AcUI0bq/1vgBXGdywmrf6AP4xGGDnxxHQ2ueQPnx5paPGKb4fgBDNwF8P6Xhh4Hx/6uf8c14/NqHTZDGXXtKhwUHTSm2LxOl778AfiISlA==).

**Comment by michalsnik** [comment]
Please try `eslint-plugin-vue@next` @adcentury and let us know if you still have this issue :)

**Comment by michalsnik** [comment]
Closing for now. Feel free to reopen if you're still having this problem.

**Comment by roland-reed** [comment]
```
<template>
    <Row class="row">
        <Col v-for="(tpl, idx) in tplList" :key="tpl.id">
            <Card
                :id="tpl.id"
                :index="idx"
            ></Card>
        </Col>
    </Row>
</template>
<script>
export default {
}
</script>
```

I can reproduce the issue with the code above. Problem will disappear when changing `<Col>` to `<div>`, which is:

```
<template>
    <Row class="row">
        <div v-for="(tpl, idx) in tplList" :key="tpl.id">
            <Card
                :id="tpl.id"
                :index="idx"
            ></Card>
        </div>
    </Row>
</template>
<script>
export default {
}
</script>
```

**Comment by roland-reed** [comment]
Here is the online [demo](https://mysticatea.github.io/vue-eslint-demo/#eJyNVstu4zgQ/BVBp13AXANzNDy5zHVPs3P0hZJaEjMUqW1Stpwg/75FPSnJhw2QwKxq9qO6m/FnmtuC0kt69dS0Wnp6u5kEP9ef9pHkWjr3/ZayfdzSiRjIH1Ynd1FaBvmHb/UpUUX/Z6JMgsPfyvlbmlx+0xM0gL9Usbk+upBcbKHwc1FFdOcVbwrqYYJ4e/7teg5O4zzPSHQu6IyK8Pl6jiq9upxV6/GJ+tayTwoqZad98nkzX8F24tNTyp0ml14+IZhxnrvcWxaua4nTy7dTCi1EoZhyr6wZkIq8JxZMvuMRMVbk0pEoCMKyDJZuIYLUAr+qWo1t00omYagSH8R2xU1xMDXOaoqP/oWJl8YP19WS5sh4thqpVtTPYEFZV1VTdcNZkydxlyuC4oXkym2AsZCGmox4y2AeYkCrHE0YFJlRNMY/RV5DnDxoN/jakq0MqpoNuBz6XcnUe5Yis1BGmhDJbxlHjZqRUmrta7ZdVS9QZ/Kdx0rbTOodqIxBsq+6qsxdalWMyrYLyjh3MBaPWnlyrcwXCRrVUyEGCOOAPnuZLe4MPYR7NpnVM2Kzd9Sl9WJicy8XlmlMilYAeUzeZ8yRLncFwYBd6C3LtWW+Vk5khEGnaO5BdFjJcj2gUGwBimiwSEorQyvHJNHdTEeQkyWJUhkU8dyhmHsZz2lnOge/Wma0FjyBmMsVQkmEISSXy3YMBZ2QlHA1lryGrAPI9G+HlRVPRXpEcFMoZ+QYcuydf7Zkx/ruHZ2xlA1hi6Zlv9PCvLs+hHZrLgFFPtI9MUjKDAvdBWlathDQK9rYYQ/LzilTieFtxV9Vxvx2i1Zw2iRsBqsM/jc8kyO+I+b+osOaAQ45WTNUJL3cGKiCBJUlynT/I/v5WQ2BtnjvEUliHvD+5DXF5NS9JQn3gtzLGZoUpBEP5etRqoWeW7pWpdyBDPkPbRXRW36wYsJcb97vmB07FJe6MMPUTP9IhliRSfD1SsvVzTR0s5hsrd9xd5GpaYJjMNdW/j6g2BSKxyjGD2Cs5IzVvtEH8IXDBt8jjobWvIDyY+SWjxiW9XEAwzQB/Dql4ZuA8f+oj/HmePz1DAufjk/qKR3eMdSU4pFlovTrP4ak/gs=).

**Comment by ota-meshi** [comment]
The `<Col>` element is recognized as an HTML `<col>` element.
Since the HTML `<col>` element is a void element, it is treated as if there are no internal elements.
Also, `</col>` will cause a parsing error as an invalid end tag.

Please rename the `<Col>` component.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/col
